url416ee5b4ee_martinmarinov_HonProj_tgz-pJ8-martin_gui_MainJ8/src/martin/experiments/ExperimentVisualizer.java

```diff
@@ -0,0 +1,123 @@
+/*******************************************************************************
+ * Copyright (c) 2013 Martin Marinov.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the GNU Public License v3.0
+ * which accompanies this distribution, and is available at
+ * http://www.gnu.org/licenses/gpl.html
+ * 
+ * Contributors:
+ *     Martin - initial API and implementation
+ ******************************************************************************/
+package martin.experiments;
+
+import java.awt.Color;
+import java.awt.Graphics;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+
+import javax.imageio.ImageIO;
+
+public class ExperimentVisualizer {
+	
+	private final static int left_offset = 110;
+	private final static int vert_offset = 1;
+	private final static int text_offset = 5;
+	private final static int resid_offset = 55;
+	private final static int width = 610;
+	private final static int height = 40;
+	
+	private final ArrayList<BufferedImage> images = new ArrayList<BufferedImage>();
+	
+	public void add(final String alabel, final Double[] a, final String blabel, final Double[] b) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 58-93 into a separate method.